tonic/src/status.rs

```diff
@@ -325,7 +325,10 @@ impl Status {
         })
     }
 
-    pub(crate) fn try_from_error(
+    /// Create a `Status` from various types of `Error`.
+    ///
+    /// Returns the error if a status could not be created.
```

Comment:
I am a bit cautious about exposing these error conversions but I would be ok if we added a note that said, the internal downcast methods/types may change and that you can't depend on it to be stable internally.

Reply:
Stable as in it might create a different status? Or as in might not work at all (i.e., returns the error)?

Reply:
Return a different status based on the input type.